reStream.sh

```diff
@@ -66,26 +78,48 @@ fi
 
 
 
-# calculte how much bytes the window is
+output_args=()
+video_filters=()
+
+# calculate how much bytes the window is
 window_bytes="$(($width*$height*$bytes_per_pixel))"
 
 # rotate 90 degrees if landscape=true
-landscape_param="$($landscape && echo '-vf transpose=1')"
+$landscape && video_filters+=('transpose=1')
 
 # read the first $window_bytes of the framebuffer
 head_fb0="dd if=/dev/fb0 count=1 bs=$window_bytes 2>/dev/null"
 
 # loop that keeps on reading and compressing, to be executed remotely
 read_loop="while $head_fb0; do $loop_wait; done | $compress"
 
+if [ "$output_path" = - ]; then
+    output_command=ffplay
+else
+    output_command=ffmpeg
+
+    if [ "$format" != - ]; then
+        output_args+=('-f' "$format")
+    fi
+
+    output_args+=("$output_path")
+fi
+
+# set frame presentation time to the time it was received
+video_filters+=('setpts=(RTCTIME - RTCSTART) / (TB * 1000000)')
+
+joined_filters=$(IFS=,; echo "${video_filters[*]}")
```

Comment:
```suggestion

```

Reply:
This is not needed anymore because in the next line I suggest using prefix substitution.